File: Pyhton/Problems/Decodestr.py
def decode(s):
    numstack=[]
    strstack=[]
    currentnum=0
    currentstr=''
    for char in s:
        if char.isdigit():
            currentnum=currentnum*10+int(char)
        elif char=='[':
            numstack.append(currentnum)
            strstack.append(currentstr)
            currentnum=0
            currentstr=''
        elif char==']':
            repeattimes=numstack.pop()
            prevtr=strstack.pop()
            currentstr=prevtr+currentstr*repeattimes
        else:
            currentstr+=char
    return currentstr

# Test the function
print(decode('3[a2[c]]'))  # Output: accaccacc

File: cricket/src/problem/Decode_str.java
package problem;
import java.util.*;

class Decode_str {

    public static String decode(String str) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);

            if (ch == ']') {
                String sub = "";
                String k = "";

                // Pop characters to form the substring until '[' is found
                while (!stack.isEmpty() && stack.peek() != '[') {
                    sub = stack.pop() + sub;
                }

                // Pop the '['
                if (!stack.isEmpty() && stack.peek() == '[') {
                    stack.pop();
                }

                // Now get the digits (number) before '['
                while (!stack.isEmpty() && Character.isDigit(stack.peek())) {
                    k = stack.pop() + k;
                }

                // Repeat the substring
                String repeated = sub.repeat(Integer.parseInt(k));

                // Push the repeated characters back to the stack
                for (int j = 0; j < repeated.length(); j++) {
                    stack.push(repeated.charAt(j));
                }
            } else {
                stack.push(ch);
            }
        }

        // Final result
        StringBuilder res = new StringBuilder();
        while (!stack.isEmpty()) {
            res.append(stack.pop());
        }

        return res.reverse().toString();
    }

    public static void main(String[] args) {
        String input = "3[a2[c]]";
        String output = decode(input);
        System.out.println(output);  // accaccacc
    }
}
